app/Helpers/PaymentService.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Helpers;
+
+use App\Models\CustomerCart;
+use App\Models\HistoryTransaction;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Ixudra\Curl\Facades\Curl;
+
+class PaymentService
+{
+
+
+    public static function CheckOut($product)
+    {
+
+        try {
+            $qty = 0;
+            $items = [];
+            $cartIds = [];
+            foreach ($product['products'] as $prod) {
+                $qty += $prod['qty'];
+                $cartIds[] = $prod['id'];
+                $items[] =      [
+                    'name' => $prod['name'],
+                    'quantity' => $prod['qty'],
+                    'description' => 'Description of the test product',
+                    'amount' => [
+                        'value' => $prod['price'],
+                        'details' => [
+                            'subtotal' => $prod['price'] * $prod['qty'],
+                            'discount' => 0,
+                            'serviceCharge' => 0,
+                            'shippingFee' => 0,
+                            'tax' => 0
+                        ]
+                    ],
+                    'totalAmount' => [
+                        'currency' => 'PHP',
+                        'value' => $prod['price'] * $prod['qty'],
+                        'details' => [
+                            'subtotal' => $prod['price'] * $prod['qty']
+                        ]
+                    ]
+                ];
+            }
+
+
+            $data = [
+                'totalAmount' => [
+                    'currency' => 'PHP',
+                    'value' => $product['total'],
+                    'details' => [
+                        'discount' => 0,
+                        'serviceCharge' => 0,
+                        'shippingFee' => 0,
+                        'tax' => 0,
+                        'subtotal' => $product['total']
+                    ]
+                ],
+                'items' => $items,
+                'redirectUrl' => [
+                    'success' => env("APP_URL") . '/success',
+                    'failure' => env("APP_URL") . '/failure',
+                    'cancel' => env("APP_URL") . '/canceled'
+                ],
+                'requestReferenceNumber' => 'ABC123',
```

Comment:
The Request Reference Number is typically uniquely generated. Kindly create a function that will generate a unique one every checkout.